v0/destinations/branch/transform.js

```diff
@@ -81,72 +59,93 @@ function getCategoryAndName(rudderEventName) {
 }
 
 function getUserData(message) {
-  const { context } = message;
-  const { os } = message.context;
+  let userData = {
+    os: get(message, "context.os.name"),
+    os_version: get(message, "context.os.version"),
+    app_version: get(message, "context.app.version"),
+    model: get(message, "context.device.model"),
+    brand: get(message, "context.device.brand"),
+    screen_dpi: get(message, "context.screen.density"),
+    screen_height: get(message, "context.screen.height"),
+    screen_width: get(message, "context.screen.width"),
+    developer_identity: getFieldValueFromMessage(message, "userId"),
+    user_agent: get(message, "context.userAgent")
+  };
 
-  if (!os || !isDefinedAndNotNullAndNotEmpty(os.name)) {
-    throw new CustomError(
-      "os name is missing in the payload and please make sure to insert it at context.os.name",
-      400
-    );
+  const os = get(message, "context.os.name");
+
+  if (isDefinedAndNotNull(os)) {
+    if (isAppleFamily(os)) {
+      userData.idfa = get(message, "context.device.advertisingId");
+      userData.idfv = get(message, "context.device.id");
+    } else if (os.toLowerCase() === "android") {
+      userData.android_id =
+        get(message, "context.device.id") || get(message, "context.android_id");
+      userData.aaid = get(message, "context.device.advertisingId");
```

Comment:
For `idfa`, `idfv` and `aaid` fields, let's keep the previous mapping as default and higher in the order.